cmd/awsDataGather/main.go

```diff
@@ -0,0 +1,110 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/red-hat-storage/ocs-osd-deployer/pkg/aws"
+	"github.com/red-hat-storage/ocs-osd-deployer/utils"
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+func main() {
+	ctrl.SetLogger(zap.New(zap.UseDevMode(true)))
+	log := ctrl.Log.WithName("main")
+
+	namespace, found := os.LookupEnv("NAMESPACE")
+	if !found {
+		fmt.Fprintf(os.Stderr,
+			"NAMESPACE environment variable not found\n")
+		os.Exit(1)
+	}
+	deploymentName, found := os.LookupEnv("DEPLOYMENT_NAME")
```

Comment:
This was on Ohad's suggestion. So that the ConfigMap doesn't go away if the pod gets deleted or restarts, only when the deployment is deleted should the ConfigMap be removed.